ARKit-QRScanner/QRScanner.swift

```diff
@@ -11,59 +11,303 @@ import ARKit
+    public static func findQRTransform(in frame: ARFrame, orientation: UIInterfaceOrientation,viewSize: CGSize) -> [QRResponse] {
+        let features = QRScanner.findQR(in: frame.capturedImage)
+        
+        let imageWidth = CGFloat(CVPixelBufferGetWidth(frame.capturedImage))//1920
+        let imageHeight = CGFloat(CVPixelBufferGetHeight(frame.capturedImage))//1440
+        
+        let factorH = viewSize.height / imageWidth
+        let factorW = viewSize.width / imageHeight
+        let factor = CGFloat.minimum(factorH, factorW)
+        
+        return features.reduce(into: [QRResponse]()) { (responses, feature) in
+            var cgpointsInImage = [feature.bottomLeft,feature.bottomRight,feature.topRight,feature.topLeft]
+            
+            
+            let bottomMiddle = CGPoint(x: (feature.bottomLeft.x + feature.bottomRight.x) * 0.5, y: (feature.bottomLeft.y + feature.bottomRight.y) * 0.5)
+            cgpointsInImage.append(bottomMiddle)
+            
+            let rightMiddle = CGPoint(x: (feature.topRight.x + feature.bottomRight.x) * 0.5, y: (feature.topRight.y + feature.bottomRight.y) * 0.5)
+            cgpointsInImage.append(rightMiddle)
+            
+            let topMiddle = CGPoint(x: (feature.topRight.x + feature.topLeft.x) * 0.5, y: (feature.topRight.y + feature.topLeft.y) * 0.5)
+            cgpointsInImage.append(topMiddle)
+            
+            let leftMiddle = CGPoint(x: (feature.bottomLeft.x + feature.topLeft.x) * 0.5, y: (feature.bottomLeft.y + feature.topLeft.y) * 0.5)
+            cgpointsInImage.append(leftMiddle)
+            
+            let center = CGPoint(x: (feature.bottomLeft.x + feature.topLeft.x + feature.topRight.x + feature.bottomRight.x) * 0.25, y: (feature.bottomLeft.y + feature.topLeft.y + feature.topRight.y + feature.bottomRight.y) * 0.25)
+            cgpointsInImage.append(center)
+            
+            
+            // convert to View coor
+            let cgpointsInView = cgpointsInImage.map { (cgpointInImage) -> CGPoint in
+                let cgpointInView = CGPoint(x: (cgpointInImage.y - imageHeight * 0.5)*factor + viewSize.width * 0.5, y: (cgpointInImage.x - imageWidth * 0.5)*factor + viewSize.height * 0.5)
+                return cgpointInView
+            }
+            
+            if cgpointsInView.count < 9 {return}
+            
+            let bottomLeftInView = cgpointsInView[0]
+            let bottomRightInView = cgpointsInView[1]
+            let topRightInView = cgpointsInView[2]
+            let topLeftInView = cgpointsInView[3]
+            let centerInView = cgpointsInView[8]
+            
+            var bottomLeftRough = simd_float3(repeating: 0)
+            var bottomRightRough = simd_float3(repeating: 0)
+            var topRightRough = simd_float3(repeating: 0)
+            var topLeftRough = simd_float3(repeating: 0)
+            
+            let points = cgpointsInImage.reduce(into: [simd_float3]()) { (points, cgpoint) in
```

Comment:
Looks like a `compactMap` would work better for you here
https://developer.apple.com/documentation/swift/sequence/2950916-compactmap